internal/tests/api/wallet_test.go

```diff
@@ -116,6 +174,126 @@ func TestWalletBalance(t *testing.T) {
+	if err != nil {
+		fail("Error creating custom snapshot: %v", err)
+	}
+	defer wallet_cleanup(snapshotName)
+
+	// Commit a block just so the latest block is fresh - otherwise the sync progress check will
+	// error out because the block is too old and it thinks the client just can't find any peers
+	err = testMgr.CommitBlock()
+	if err != nil {
+		t.Fatalf("Error committing block: %v", err)
+	}
+
+	// Regen the wallet
+	apiClient := testMgr.GetApiClient()
+	derivationPath := string(wallet.DerivationPath_Default)
+	index := uint64(0)
+	_, err = apiClient.Wallet.Recover(&derivationPath, keys.DefaultMnemonic, &index, goodPassword, true)
+	require.NoError(t, err)
+	t.Log("Recover called")
+
+	message := []byte("hello world")
+	response, err := apiClient.Wallet.SignMessage(message)
+	require.NoError(t, err)
+	t.Log("SignMessage called")
+
+	require.NotEmpty(t, response.Data.SignedMessage)
+	prefix := fmt.Sprintf("\x19Ethereum Signed Message:\n%d", len(message))
+	msgHash := crypto.Keccak256Hash([]byte(prefix + string(message)))
+	sig := response.Data.SignedMessage
+
+	// Recover the public key from the signature
+	r, s, v := sig[:32], sig[32:64], sig[64]
+	pubKey, err := crypto.SigToPub(msgHash.Bytes(), append(r, append(s, v-27)...))
+	require.NoError(t, err)
+
+	// Make sure that the recovered address is the signer address
+	recoveredAddr := crypto.PubkeyToAddress(*pubKey)
+	require.Equal(t, expectedWalletAddress, recoveredAddr)
+	t.Logf("Successfully signed message")
+
+}
+
+func TestWalletSend_EthSuccess(t *testing.T) {
+	// Take a snapshot, revert at the end
+	snapshotName, err := testMgr.CreateCustomSnapshot(osha.Service_EthClients | osha.Service_Filesystem)
+	if err != nil {
+		fail("Error creating custom snapshot: %v", err)
+	}
+	defer wallet_cleanup(snapshotName)
+
+	// Commit a block just so the latest block is fresh - otherwise the sync progress check will
+	// error out because the block is too old and it thinks the client just can't find any peers
+	err = testMgr.CommitBlock()
+	if err != nil {
+		t.Fatalf("Error committing block: %v", err)
+	}
+
+	// Regen the wallet
+	apiClient := testMgr.GetApiClient()
+	derivationPath := string(wallet.DerivationPath_Default)
+	index := uint64(0)
+	_, err = apiClient.Wallet.Recover(&derivationPath, keys.DefaultMnemonic, &index, goodPassword, true)
+	require.NoError(t, err)
+	t.Log("Recover called")
+
+	targetAddress := common.HexToAddress("0x95222290dd7278aa3ddd389cc1e1d165cc4bafe5")
+	response, err := apiClient.Wallet.Send(eth.EthToWei(1), "eth", targetAddress)
+	require.NoError(t, err)
+	t.Log("Send called")
+
+	require.Equal(t, targetAddress, response.Data.TxInfo.To)
+	require.Equal(t, eth.EthToWei(1), response.Data.TxInfo.Value)
+	require.NotEmpty(t, response.Data.TxInfo.SimulationResult)
+
+	require.True(t, response.Data.CanSend)
+	require.False(t, response.Data.InsufficientBalance)
+	t.Logf("Successfully sent ETH")
```

Comment:
This doesn't actually send the ETH, this just creates a transaction and simulates it. I assume based on the name you want to actually send it; in that case you'll need to do something like this:

```go
	sub, _ := eth.CreateTxSubmissionFromInfo(response.Data.TxInfo, nil)
	submitResponse, err := apiClient.Tx.SubmitTx(sub, nil, eth.GweiToWei(10), eth.GweiToWei(1))
	require.NoError(t, err)
	t.Log("SubmitTx called")
	
	err = testMgr.CommitBlock()
	require.NoError(t, err)

	_, err = apiClient.Tx.WaitForTransaction(submitResponse.Data.TxHash)
	require.NoError(t, err)
	t.Log("Waiting complete")
```

Then use the EC binding to check the balance of `targetAddress` and the node wallet to make sure they increased & decreased properly.

Reply:
fixed